pctest/test_publish_websocket.cpp

```diff
@@ -0,0 +1,268 @@
+#include <stdlib.h>
+#include <unistd.h>
+#include <iostream>
+#include <ctime>
+#include <iostream>
+#include <memory>
+#include <list>
+
+#include <jcon/json_rpc_tcp_client.h>
+#include <jcon/json_rpc_websocket_client.h>
+
+#include <QCoreApplication>
+#include <QDebug>
+#include <QTimer>
+#include <QUrl>
+#include <QtGlobal>
+
+/*
+This file demonstrates how to use publish price updates to the Pyth Network using
+the pythd websocket API described at https://docs.pyth.network/publish-data/pyth-client-websocket-api.
+
+High-level flow:
+- Call get_product_list to fetch the product metadata, enabling us to associate the price account public 
+  keys with the symbols we want to publish for.
+- For each price account public key, call subscribe_price_sched to subscribe to the price update schedule
+  for that price account. This will return a subscription ID.
+- When we receive a notify_price_sched message, we should:
+  - Look up the price account associated with the subscription ID.
+  - If we have any new price information for that price account, send an update_price message to submit this
+    to the network.
+*/
+
+// This class is responsible for the communication with pythd over the websocket protocol
+// described at https://docs.pyth.network/publish-data/pyth-client-websocket-api
+// 
+// It abstracts away any solana-specific details, such as public keys.
+class pythd_websocket
```

Comment:
can we expose this as non-test code so that people can use it without copy/pasting ?

Reply:
Decided against this, as

- each publisher is likely to use the api slightly differently
- publishers are unlikely to use QT as the event loop